prometheus/src/main/scala/filodb/prometheus/ast/Vectors.scala

```diff
@@ -139,9 +138,30 @@ sealed trait Vector extends Expression {
   val regexColumnName: String = "::(?=[^::]+$)" //regex pattern to extract ::columnName at the end
 
   // Convert metricName{labels} -> {labels, __name__="metricName"} so it's uniform
-  lazy val mergeNameToLabels: Seq[LabelMatch] = {
+  // Note: see makeMergeNameToLabels before changing the laziness.
+  lazy val mergeNameToLabels: Seq[LabelMatch] = makeMergeNameToLabels()
+
+  /**
+   * Constructs the mergeNameToLabels member.
+   *
+   * Note:
+   * At present, FiloDB requires the existence of a metric name for all vector selectors, and
+   *   that requirement is enforced at the initialization of mergeNameToLabels. However,
+   *   this makes it impossible to initialize a Vector type with PromQL-valid label selectors
+   *   that lack an implicit/explicit __name__ label.
+   *
+   * If you need to make use of a name-optional Vector (i.e. as a temporary vessel for parsed PromQL),
+   *   the hacky workaround is as follows:
+   *     (1) lazily initialize mergeNameToLabels so a metric name is not explicitly required
+   *         at the Vector's initialization, and
+   *     (2) when needed, locally initialize a "mergeNameToLabels-like" value without
+   *         a name requirement.
+   *
+   * This method exists only to consolidate code for the initialization of these values.
+   */
```

Comment:
I've been riffing off this idea on a local branch, but I keep running into the same two problems:
1. There is no guarantee that the validation will ever run.
2. The validation can't be manually run for tests (especially when `InstantExpression`s are nested inside of other `Expression` types) without spaghetti code.

Please, please, please see TODO:PR for a much safer fix. I'll add comments to argue my case there.

Reply:
This branch still isn't as clean/straightforward as I had hoped-- let the record show that I still think #1314 is substantially safer.